sql/core/src/main/scala/org/apache/spark/sql/execution/stat/StatFunctions.scala

```diff
@@ -264,7 +264,10 @@ object StatFunctions extends Logging {
     }
 
     val selectedCols = ds.logicalPlan.output
-      .filter(a => a.dataType.isInstanceOf[NumericType] || a.dataType.isInstanceOf[StringType])
+      .filter(a => a.dataType.isInstanceOf[NumericType]
+        || a.dataType.isInstanceOf[StringType]
+        || a.dataType.isInstanceOf[DateType]
```

Comment:
Yes, let's write a UT. Does it work BTW? Looks at least `mean` and date type won't work here.

Reply:
I'm working on getting the test suite running on my machine, so I need some time. I don't think that the `mean` will be the issue, this is just the element in the middle of the sorted collection, however, the `stddev` will be tricky. For the StringType this is just `null`.